src/main/java/Problem1.java

```diff
@@ -1,8 +1,37 @@
 public class Problem1 {
 
+    private static int lowAdjust = 0;
```

Comment:
use a helper function for recursion. this way there's no need to add member variable.